src/components/views/elements/Quote.js

```diff
@@ -0,0 +1,155 @@
+
+const MILLIS_IN_DAY = 86400000;
+function wantsDateSeparator(parentEvent, event) {
+    const parentEventDate = parentEvent.getDate();
+    const eventDate = event.getDate();
+    if (!eventDate || !parentEventDate) {
+        return false;
+    }
+    // Return early for events that are > 24h apart
+    if (Math.abs(parentEvent.getTs() - eventDate.getTime()) > MILLIS_IN_DAY) {
+        return true;
+    }
+
+    // Compare weekdays
+    return parentEventDate.getDay() !== eventDate.getDay();
+}
+
+export default class Quote extends React.Component {
+    static isMessageUrl(url) {
+        return !!REGEX_LOCAL_MATRIXTO.exec(url);
+    }
+
+    static childContextTypes = {
+        matrixClient: PropTypes.object,
+    };
+
+    static propTypes = {
+        // The matrix.to url of the event
+        url: PropTypes.string,
+        // The parent event
+        parentEv: PropTypes.instanceOf(MatrixEvent),
+        // Whether this isn't the first Quote, and we're being nested
+        isNested: PropTypes.bool,
+    };
+
+    constructor(props, context) {
+        super(props, context);
+
+        this.state = {
+            // The event related to this quote
+            event: null,
+            show: !this.props.isNested,
+        };
+
+        this.onShowNested = this.onShowNested.bind(this);
+    }
+
+    getChildContext() {
+        return {
+            matrixClient: MatrixClientPeg.get(),
+        };
+    }
+
+    componentWillReceiveProps(nextProps) {
+        let roomId;
+        let prefix;
+        let eventId;
+
+        if (nextProps.url) {
+            // Default to the empty array if no match for simplicity
+            // resource and prefix will be undefined instead of throwing
+            const matrixToMatch = REGEX_LOCAL_MATRIXTO.exec(nextProps.url) || [];
+
+            roomId = matrixToMatch[1]; // The room ID
+            prefix = matrixToMatch[2]; // The first character of prefix
+            eventId = matrixToMatch[3]; // The event ID
+        }
+
+        const room = prefix === '#' ?
+            MatrixClientPeg.get().getRooms().find((r) => {
+                return r.getAliases().includes(roomId);
+            }) : MatrixClientPeg.get().getRoom(roomId);
+
+        // Only try and load the event if we know about the room
+        // otherwise we just leave a `Quote` anchor which can be used to navigate/join the room manually.
+        if (room) this.getEvent(room, eventId);
+    }
+
+    componentWillMount() {
+        this.componentWillReceiveProps(this.props);
+    }
+
+    async getEvent(room, eventId) {
+        let event = room.findEventById(eventId);
+        if (event) {
+            this.setState({room, event});
+            return;
+        }
+
+        await MatrixClientPeg.get().getEventTimeline(room.getUnfilteredTimelineSet(), eventId);
+        event = room.findEventById(eventId);
+        this.setState({room, event});
+    }
+
+    onShowNested() {
```

Comment:
Current convention is generally that the onThing is what the user did rather than what action that should cause, ie. onQuoteClick in this case